aem-authoring-toolkit-assets/src/main/content/jcr_root/apps/authoring-toolkit/depends-on/js/additional-actions/getPropertyDependsOnAction.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * @author Alexey Stsefanovich (ala'n), Liubou Masiuk (liubou-masiuk), Yana Bernatskaya (YanaBr)
+ * @version 2.2.4
+ * Custom action to get component property
+ * property path can be relative (e.g. 'node/nestedProperty' or '../../parentCompProperty')
+ *
+ * query - current node (this)
+ * */
+
+(function (Granite, $, DependsOn) {
+    'use strict';
+
+    const PARENT_DIR_REGEX = /\.\.\//g;
+
+    function getLevel(path) {
+        const dirMatches = path.match(PARENT_DIR_REGEX);
+        return dirMatches && dirMatches.length || 0;
+    }
+
+    const RequestCache = (function (timeout) {
+        let clearTm = null;
+        let map = new Map();
+
+        function clearCache() {
+            map.clear();
+            console.debug('[DependsOn] Custom action "get-property" cache cleared.');
+        }
+
+        function get(url) {
+            url = Granite.HTTP.externalize(url);
+            if (!map.has(url)) map.set(url, $.get(url));
+            if (clearTm) clearTimeout(clearTm);
+            clearTm = setTimeout(clearCache, timeout);
+            return map.get(url);
+        }
+
+        return {get: get, clear: clearCache};
+    })(2000);
+
+    function getParentProperty(currentResource, options) {
```

Comment:
Let's try to remove query part (resolving of dialog path and current Resource) into internal details.
And move property path to query.